java/java-psi-api/src/com/intellij/psi/util/ClassUtil.java

```diff
@@ -225,6 +233,31 @@ private static boolean isIndexed(String name) {
     return aClass.getQualifiedName();
   }
 
+  /**
+   * Returns the binary class name, i.e. the string that would be returned by {@link Class#getName}.
+   *
+   * <p>Inner classes ({@code Foo$Bar}), anonymous ({@code Foo$1}), and local ({@code Foo$1Bar}) classes are taken into account.
+   */
+  public static @Nullable @NlsSafe String getBinaryClassName(@NotNull PsiClass aClass) {
```

Comment:
Huge thanks for writing the Javadoc. Minor suggestion:

I know `Class#getName` refers to JLS too but I think it's useful to surface this info early.

```suggestion
  /**
   * Returns the binary class name, i.e. the string that would be returned by {@link Class#getName}. See JLS 13.1.
   *
   * <p>Inner ({@code Foo$Bar}), anonymous ({@code Foo$1}), and local ({@code Foo$1Bar}) classes are taken into account.
   */
  public static @Nullable @NlsSafe String getBinaryClassName(@NotNull PsiClass aClass) {
```